File: C++/383-ransom-note.cpp
class Solution {
public:
  bool canConstruct(string ransomNote, string magazine) {
    vector<int> h(26);

    for (const auto& c : magazine) {
      h[c - 'a']++;
    }

    for (const auto& c : ransomNote) {
      h[c - 'a']--;
    }

    for (const auto& c : h) {
      if (c < 0) return false;
    }
    
    return true;
  }
};

File: archive/JavaScript/383-ransom-note.js
let canConstruct = (ransomNote, magazine) => {
  let m = {}
  for (let b of magazine) {
    if (!m.hasOwnProperty(b)) {
      m[b] = 0
    }
    m[b]++
  }
  for (let b of ransomNote) {
    if (!m.hasOwnProperty(b) || m[b] == 0) {
      return false
    }
    m[b]--
  }
  return true
}
